flink-container/src/test/java/org/apache/flink/container/entrypoint/StandaloneJobClusterConfigurationParserFactoryTest.java

```diff
@@ -73,17 +96,37 @@ public void testEntrypointClusterConfigurationParsing() throws FlinkParseExcepti
 
 		assertThat(clusterConfiguration.getSavepointRestoreSettings(), is(equalTo(SavepointRestoreSettings.none())));
 
-		assertThat(clusterConfiguration.getJobId(), is(equalTo(DEFAULT_JOB_ID)));
+		assertThat(clusterConfiguration.getJobId(), is(equalTo(jobId)));
+	}
+
+	@Test
+	public void testJobIDDefaultsToRandomJobIdWithoutHA() throws FlinkParseException {
+		final String[] args = {"--configDir", confDirPath};
+
+		final StandaloneJobClusterConfiguration clusterConfiguration = commandLineParser.parse(args);
+
+		assertThat(clusterConfiguration.getJobId(), is(not(nullValue())));
+		assertThat(clusterConfiguration.getJobId(), is(not(equalTo(ZERO_JOB_ID))));
+	}
+
+	@Test
+	public void testJobIDDefaultsToZeroJobIdWithHA() throws FlinkParseException, FileNotFoundException {
+		final String[] args = {"--configDir", confDirPath};
+
+		final StandaloneJobClusterConfiguration clusterConfiguration = commandLineParser.parse(args);
+
+		configureHighAvailabilityGlobally();
```

Comment:
this should be the first statement in the test

Reply:
Douh, but test does not exist anymore in new version.